kover-maven-plugin/src/main/kotlin/kotlinx/kover/maven/plugin/mojo/VerifyMojo.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.maven.plugin.mojo
+
+import kotlinx.kover.features.jvm.*
+import kotlinx.kover.maven.plugin.MavenBound
+import kotlinx.kover.maven.plugin.MavenRule
+import kotlinx.kover.maven.plugin.MavenRuleFilters
+import kotlinx.kover.maven.plugin.mojo.abstracts.AbstractCoverageTaskMojo
+import org.apache.maven.plugin.MojoFailureException
+import org.apache.maven.plugins.annotations.LifecyclePhase
+import org.apache.maven.plugins.annotations.Mojo
+import org.apache.maven.plugins.annotations.Parameter
+import java.io.File
+
+/**
+ * Mojo to verify coverage by specified rules.
+ */
+@Mojo(name = "verify", defaultPhase = LifecyclePhase.VERIFY)
+class VerifyMojo : AbstractCoverageTaskMojo() {
+    @Parameter(property = "kover.warningInsteadOfFailure", defaultValue = "false", required = true)
+    private var warningInsteadOfFailure: Boolean = false
+
+    @Parameter
```

Comment:
I wonder how this works. Is it created and inserted by Maven itself? How it can modify a private val? Can data classes like `MavenRule` be created with respect to default values?

Reply:
> How it can modify a private val?

~~magic~~ reflection :)

> Is it created and inserted by Maven itself?

Yes, maven have [special mappers](https://maven.apache.org/guides/mini/guide-configuring-plugins.html#Mapping_Complex_Objects) for composite types.

> Can data classes like MavenRule be created with respect to default values?
It works quite simply, but most often in such mappers you need constructors without parameters.
if this parameter is not specified in the configuration, then the value with which the field is initialized in `<init>` will be